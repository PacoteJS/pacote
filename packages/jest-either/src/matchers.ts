export interface AsymmetricMatcher {
  asymmetricMatch(other: any): boolean
}

export function isAsymmetricMatcher(
  matcher: any
): matcher is AsymmetricMatcher {
  return typeof matcher.asymmetricMatch === 'function'
}
