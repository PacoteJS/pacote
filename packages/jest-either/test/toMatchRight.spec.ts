import { left, right } from 'fp-ts/lib/Either'
import matchers from '../src'

expect.extend(matchers)

describe('.toMatchRight()', () => {
  test('passes when object matches the right', () => {
    expect(right({ a: 1, b: 2 })).toMatchRight({ a: 1 })
  })

  test('fails when object does not match the left', () => {
    expect(() =>
      expect(right({ a: 1 })).toMatchRight({ a: 2 })
    ).toThrowErrorMatchingSnapshot()
  })

  test('fails when object is a left', () => {
    expect(() =>
      expect(left({ a: 1 })).toMatchRight({ a: 1 })
    ).toThrowErrorMatchingSnapshot()
  })
})

describe('.not.toMatchRight()', () => {
  test('passes when object is a left', () => {
    expect(left({ a: 1 })).not.toMatchRight({ a: 1 })
  })

  test('passes when object does not match the right', () => {
    expect(right({ a: 1 })).not.toMatchRight({ a: 2 })
  })

  test('fails when object matches the right', () => {
    expect(() =>
      expect(right({ a: 1, b: 2 })).not.toMatchRight({ a: 1 })
    ).toThrowErrorMatchingSnapshot()
  })
})
