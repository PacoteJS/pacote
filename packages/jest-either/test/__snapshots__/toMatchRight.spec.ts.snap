// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toMatchRight() fails when object matches the right 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).not.toMatchRight([22m[32mrightMatch[39m[2m)[22m

Expected Either not to match right:
  [31m{\\"a\\": 1}[39m

Received right:
  [31m{\\"a\\": 1, \\"b\\": 2}[39m"
`;

exports[`.toMatchRight() fails when object does not match the left 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toMatchRight([22m[32mrightMatch[39m[2m)[22m

Expected Either to match right:
  [31m{\\"a\\": 2}[39m

Received right:
  [31m{\\"a\\": 1}[39m"
`;

exports[`.toMatchRight() fails when object is a left 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toMatchRight([22m[32mrightMatch[39m[2m)[22m

Expected Either to match right:
  [31m{\\"a\\": 1}[39m

Received left."
`;
